### PR TITLE
Add a few test cases to ReceiveBlockNoPubsub with race detection

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Other -- code health

**What does this PR do? Why is it needed?**

The blockchain package has very poor test coverage. This PR adds tests to ReceiveBlockNoPubsub.

**Which issues(s) does this PR fix?**

N/A

**Other notes for review**

~Also fixes a data race in state!~ #6322 

Removes  unused interface and relevant dead code for ParticipationFetcher.

### CODE REVIEW

File: beacon-chain/blockchain/receive_block_test.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+			check: func(t *testing.T, s *Service) {
+				t.Skip("TODO: Need to add a better mock for state notifier.")
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			db := testDB.SetupDB(t)
+			defer testDB.Cleanup(t, db)
```
Comment (farazdagi) [inline beacon-chain/blockchain/receive_block_test.go:129]: In `testDB.SetupDB()` we register cleanup function with `t.Cleanup()` - so no need to defer call it manually. Just calling `db := testDB.SetupDB()` is enough.

File: beacon-chain/db/testing/cleanup_db.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+// Cleanup clears and closes a database.
+func Cleanup(t *testing.T, database db.Database) {
+	if err := database.ClearDB(); err != nil {
+		t.Error(err)
+	}
+	if err := database.Close(); err != nil {
+		t.Error(err)
+	}
+}
```
Comment (farazdagi) [inline beacon-chain/db/testing/cleanup_db.go:17]: Consider updating `t.Cleanup` method in `beacon-chain/db/testing/setup_db` (if you think that procedure there doesn't clean the DB properly). There is no reason to re-introduce cleanup method that manually will be defer called (one can always limit scope using `r.Run` and within that scope call `t.SetupDB()` which will register cleanup handler).

File: beacon-chain/blockchain/receive_block_test.go (hunk 1 above)
Comment (prestonvanloon) [reply beacon-chain/blockchain/receive_block_test.go:129]: Oh great! Thanks, i'll remove the new code I added and review the implementation with t.Cleanup to ensure it's the desired behavior.

**Comment by terencechain** [comment]
@prestonvanloon the latest error is `duplicate metrics collector registration attempted`

I fixed the last error by:
1.) Remove new state mgmt condition in SaveHeadBlockRoot
2.) Reuse state summary cache between db and service

A good follow up item from this is to break `SetupDB(t testing.TB)` -> `SetupDB(t testing.TB, stateSummaryCache cache)`. The current test db set up is too strict, it doesn't allow sharing state summary cache with other services.

File: beacon-chain/blockchain/receive_block_test.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+		},
+		{
+			name: "sets epochParticipation",
+			args: args{
+				block: genFullBlock(t, testutil.DefaultBlockGenConfig(), 1 /*slot*/),
+			},
+			check: func(t *testing.T, s *Service) {
+				e := s.epochParticipation[0]
+				if e == nil {
+					t.Error("epoch participation is nil")
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/receive_block_test.go:118]: Blocked on this. How can I test that epoch participation is set correctly? @terencechain

File: beacon-chain/blockchain/receive_block_test.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
+package blockchain
+
+import (
+	"context"
+	"testing"
+
+	testDB "github.com/prysmaticlabs/prysm/beacon-chain/db/testing"
+
```
Comment (terencechain) [inline beacon-chain/blockchain/receive_block_test.go:8]: goimport

Comment (prestonvanloon) [reply beacon-chain/blockchain/receive_block_test.go:8]: this is valid for goimport, I'll move it to the second group though

**Comment by prestonvanloon** [comment]
Looks like a build or test failure... fixing

File: beacon-chain/blockchain/chain_info.go [hunk 5]
```diff
@@ -68,12 +67,6 @@ type FinalizationFetcher interface {
 	PreviousJustifiedCheckpt() *ethpb.Checkpoint
 }
 
-// ParticipationFetcher defines a common interface for methods in blockchain service which
-// directly retrieves validator participation related data.
-type ParticipationFetcher interface {
```
Comment (0xKiwi) [inline beacon-chain/blockchain/chain_info.go:73]: Why was all of this removed?

Comment (prestonvanloon) [reply beacon-chain/blockchain/chain_info.go:73]: It was unused, see notes in PR description.

**Pull request merged**